### ISSUE TITLE
Can't view downloaded chapter, "File name too long" error

### ISSUE THREAD

**Issue by w4tchdoge** [mihonapp/mihon#279]
### Steps to reproduce

1. Download a chapter of a work which has a long filename.
2. Try and open the chapter.

### Expected behavior

Chapter opens and I can view it.

### Actual behavior

A toast notification pops up saying "File name too long".

### Mihon version

0.16.2

### Android version

Android 14

### Device

Samsung Galaxy Tab S8+

### Other details

I had first experienced this issue in TachiyomiSY after a recent update where storage related settings were changed. When I asked about this in the `tachiyomi-sy` support channel I was advised to see if I could reproduce this behaviour on Mihon, which I was able to.
The following text is the message I had sent in the `tachiyomi-sy` support channel after I was able to reproduce the issue in Mihon 0.16.2 (stable)

***

I can reproduce this in Mihon 0.16.2, but I had to move the folder containing the .cbz downloaded by SY to the Mihon local folder and try to open the local source
I suspest this is because the external extension has different .cbz naming/foldering than the one built-in source SY has
(extenal save the .cbz as `{full gallery name}/Chapter.cbz` while SY saves it as `{full gallery name}/{gallery id}_{version}_{full gallery name}.cbz`)
if it helps, the `{full gallery name}` in this case is 225 characters long, including spaces

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] I have gone through the [FAQ](https://mihon.app/docs/faq/general) and [troubleshooting guide](https://mihon.app/docs/guides/troubleshooting/).
- [X] I have updated the app to version **[0.16.2](https://github.com/mihonapp/mihon/releases/latest)**.
- [X] I have updated all installed extensions.
- [X] I will fill out all of the requested information in this form.

**Comment by infyProductions** [comment]
I tried to recreate this scenario to try long file names in both files and folders using local source to easily access the files in the file explorer.

Note:
- Android's internal storage uses ext4 while sdcards can be formatted in either fat32/vfat (up to 32 GBs) or exfat (64 GBs and above)
- I used an emulator (WSA) to simplify the process, assuming the storage location is on the internal stoage (/storage/emulated/0/Mihon)

Folder Test (255 chars):
- The folder's name is called "`This is a sample test folder to test long file names are supported with the mihon app specifically decompressed from a comic book archive with a different storage location. If this is fat 32-the limit is 255 characters. Applies to the same  ext4 file sys`"
Conclusion:
![image](https://github.com/mihonapp/mihon/assets/31158494/042cd1de-7a46-4027-8b94-1bd0c9a3398c)

- Mihon successfully opens the folder version and you can easily view them

File Test (253 chars for file name, 3 characters for the file extension making it 256 chars)
- The file name is called "`This is a sample test file to test long file names are supported with the mihon app specifically compressed ino a comic book archive with a different storage location. If this is fat 32-the limit is 255 characters. Also applies with - ext4 file system.cbz`" (ignore the typo just to fill it to the max limit)
Conclusion:
![image](https://github.com/mihonapp/mihon/assets/31158494/44c4094e-7c9e-4546-a4dc-d364dca536af)

- Mihon won't open the file due to the "File name too long" error.
- External programs like FX file explorer can open the archive without any errors.

Additional notes:
- It requires testing for external storages (typically SD card) formatted in the common file systems (fat32 and exfat)

**Comment by infyProductions** [comment]
While I was researching the limits for the fat32 file system, I stumbled this from a MS website of all places.

![image](https://github.com/mihonapp/mihon/assets/31158494/56364daa-6e3d-4d7d-8c08-add6a7fde32a)

Reference: https://learn.microsoft.com/en-us/windows/win32/fileio/filesystem-functionality-comparison#limits

TL;DR: the folder path must not exceed 255 characters.

Possible proposal(?): If the file and folder path (including file extensions) is too long, attempt to shorten the file name (around 250 chars) to include an underscore and the preferred file format something like this:
`<250 chars>_.png`
